jdaviz/configs/imviz/tests/test_astrowidgets_api.py

```diff
@@ -65,6 +65,59 @@ def test_center_offset_sky(self):
             self.imviz.offset_by(dsky, dsky)
 
 
+class TestZoom(BaseImviz_WCS_NoWCS):
+
+    @pytest.mark.parametrize('val', (0, -0.1, 'foo', [1, 2]))
+    def test_invalid_zoom_level(self, val):
+        with pytest.raises(ValueError, match='Unsupported zoom level'):
+            self.imviz.zoom_level = val
+
+    def test_invalid_zoom(self):
+        with pytest.raises(ValueError, match='zoom only accepts int or float'):
+            self.imviz.zoom('fit')
+
+    def assert_zoom_results(self, zoom_level, x_min, x_max, y_min, y_max, dpix):
+        assert_allclose(self.imviz.zoom_level, zoom_level)
+        assert_allclose((self.viewer.state.x_min, self.viewer.state.x_max,
+                         self.viewer.state.y_min, self.viewer.state.y_max),
+                        (x_min + dpix, x_max + dpix,
+                         y_min + dpix, y_max + dpix))
+
+    @pytest.mark.parametrize('is_offcenter', (False, True))
+    def test_zoom(self, is_offcenter):
+        if is_offcenter:
+            self.imviz.center_on((0, 0))
+            dpix = -4.5
+        else:
+            self.imviz.center_on((4.5, 4.5))
+            dpix = 0
+
+        self.assert_zoom_results(10, -0.5, 9.5, -0.5, 9.5, dpix)
+
+        # NOTE: Not sure why X/Y min/max not exactly the same as aspect ratio 1
+        self.imviz.zoom_level = 1
+        self.assert_zoom_results(1, -46, 54, -45.5, 54.5, dpix)
```

Comment:
Does seem weird some x/y min/max not matching exactly for square data and display at aspect ratio one, but that adjustment is deep within glue core and I don't feel like digging no more. If @astrofrog has ideas, then great. Otherwise, I'll just leave it be.